controllers/openstackcluster_controller.go

```diff
@@ -386,14 +393,19 @@ func reconcileBastion(scope scope.Scope, cluster *clusterv1.Cluster, openStackCl
 	return nil
 }
 
-func bastionToInstanceSpec(openStackCluster *infrav1.OpenStackCluster, clusterName string) *compute.InstanceSpec {
+func bastionToInstanceSpec(openStackCluster *infrav1.OpenStackCluster, clusterName string, computeService *compute.Service) (*compute.InstanceSpec, error) {
 	name := fmt.Sprintf("%s-bastion", clusterName)
+
+	imageID, err := computeService.GetImageID(openStackCluster.Spec.Bastion.Instance.Image)
```

Comment:
I've put that here because if the user changes the bastion spec with a new image, the instanceSpec will change and the bastion will be re-created.